src/main.c

```diff
@@ -182,8 +182,13 @@ main (int    argc,
   ret = 0;
 
  out:
-  if (sigint_id > 0)
-    g_source_remove (sigint_id);
+  if (sigint_source)
+    {
+      if( !g_source_is_destroyed(sigint_source))
```

Comment:
Just a nitpick -- this should be ``(! g_source_is_destroyed...)`` as that's how such conditions are written all over the codebase.